app/src/main/java/org/liberty/android/fantastischmemo/ui/CardMCEditor.java

```diff
@@ -0,0 +1,142 @@
+    private EditText mcOption4;
+
+    private RadioButton option1Radio;
+    private RadioButton option2Radio;
+    private RadioButton option3Radio;
+    private RadioButton option4Radio;
+
+    private Button save;
+    public static String EXTRA_DBPATH_MC = "dbpath";
+    String dbPath;
+    private AnyMemoDBOpenHelper dbOpenHelper;
+
+    private MultipleChoiceCardDao multipleChoiceCardDao;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState){
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.card_mc_editor_layout);
+
+        dbPath = getIntent().getExtras().getString(EXTRA_DBPATH_MC);
+
+        mcQuestion = (EditText) findViewById(R.id.edit_dialog_question_mc_entry);
+
+        mcOption1 = (EditText) findViewById(R.id.edit_dialog_option1_entry);
+
+        mcOption2 = (EditText) findViewById(R.id.edit_dialog_option2_entry);
+
+        mcOption3 = (EditText) findViewById(R.id.edit_dialog_option3_entry);
+
+        mcOption4 = (EditText) findViewById(R.id.edit_dialog_option4_entry);
+
+        option1Radio = (RadioButton) findViewById(R.id.op_radio_1);
+
+        option2Radio = (RadioButton) findViewById(R.id.op_radio_2);
+
+        option3Radio = (RadioButton) findViewById(R.id.op_radio_3);
+
+        option4Radio = (RadioButton) findViewById(R.id.op_radio_4);
+
+        save = (Button) findViewById(R.id.save_mc);
+        save.setOnClickListener(saveOnClickListener);
+        dbOpenHelper = AnyMemoDBOpenHelperManager.getHelper(getApplicationContext(), dbPath);
+        multipleChoiceCardDao = dbOpenHelper.getMultipleChoiceDao();
+        multipleChoiceCardDao.setHelper(dbOpenHelper);
+        updateViews();
+    }
+
+    protected AnyMemoDBOpenHelper getDbOpenHelper() {
+        return dbOpenHelper;
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        AnyMemoDBOpenHelperManager.releaseHelper(dbOpenHelper);
+    }
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+    }
+
+    public void updateViews(){
+
+        String questionMCCard = mcQuestion.getText().toString();
+        String option1MCCard = mcOption1.getText().toString();
+        String option2MCCard = mcOption2.getText().toString();
+        String option3MCCard = mcOption3.getText().toString();
+        String option4MCCard = mcOption4.getText().toString();
+
+        currentMCCard.setQuestion(questionMCCard);
+        currentMCCard.setOption1(option1MCCard);
+        currentMCCard.setOption2(option2MCCard);
+        currentMCCard.setOption3(option3MCCard);
+        currentMCCard.setOption4(option4MCCard);
+
+        if(option1Radio.isChecked()){
+            currentMCCard.setAnswer(option1MCCard);
+        }else if(option2Radio.isChecked()){
+            currentMCCard.setAnswer(option2MCCard);
+        }else if(option3Radio.isChecked()){
+            currentMCCard.setAnswer(option3MCCard);
+        }else{
```

Comment:
If the user doesn't click any of the radio buttons, like lets say they forget, will it by default set it to the last one?

Reply:
yep the default is the last one

Reply:
hmm maybe later instead of setting the last one by default when the user tries to save we can have a pop up telling them they didn't choose one!

Reply:
okie we'll add that check in the next sprint!